editoast/src/models/train_schedule.rs

```diff
@@ -47,6 +53,26 @@ pub struct TrainSchedule {
     pub timetable_id: i64,
 }
 
+impl TrainSchedule {
+    pub async fn get_by_rolling_stock_id(db_pool: Data<DbPool>, rs_id: i64) -> Result<Vec<Self>> {
+        block(move || {
+            use crate::tables::osrd_infra_trainschedule::dsl::*;
+            let mut conn = db_pool.get()?;
+            Ok(osrd_infra_trainschedule
+                .filter(rolling_stock_id.eq(rs_id))
```

Comment:
By doing so, if `rs_id` links to a valid rolling stock that has no train schedule, the function returns an empty Vec as expected. But if `rs_id` is not a valid rolling stock it still returns an empty Vec instead of an error.

Reply:
The function isn’t used, I replaced it by the sql query. I’ll remove it.